Hast.Abstractions/Hast.Vitis.Abstractions/Services/AzureHardwareImplementationComposerBuildProvider.cs

```diff
@@ -151,39 +155,26 @@ public void AddShortcutsToOtherProviders(IEnumerable<IHardwareImplementationComp
             }
         }
 
-        private async Task<T> GetResponseAsync<T>(Uri url, object post)
-            where T : AzureResponseData
-        {
-            const string jsonContentType = "application/json";
-
-            var postJson = JsonConvert.SerializeObject(post);
-            var postContent = new StringContent(postJson, Encoding.UTF8, jsonContentType);
-
-            using var client = new HttpClient();
-            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(jsonContentType));
-            var response = await client.PostAsync(url, postContent);
-
-            var content = await response.Content.ReadAsStringAsync();
-            if (string.IsNullOrWhiteSpace(content) || !(JsonConvert.DeserializeObject<T>(content) is { } result))
-            {
-                throw response.IsSuccessStatusCode
-                    ? new InvalidOperationException("Response is empty.")
-                    : new WebException(response.StatusCode.ToString());
-            }
-
-            var hasErrorMessage = !string.IsNullOrWhiteSpace(result.ErrorMessage);
-            if (!hasErrorMessage && response.IsSuccessStatusCode) return result;
-
-            _logger.LogError(
-                "There was an error in the response to the request for {0}. (request: {1}; response: {2})",
-                url,
-                postJson,
-                JsonConvert.SerializeObject(result));
-            throw hasErrorMessage
-                ? new InvalidOperationException(result.ErrorMessage)
-                : new WebException(response.StatusCode.ToString());
-        }
-
         private static string UpdateBinaryPath(string input) => Regex.Replace(input, @"\.xclbin$", ".azure.xclbin");
+
+        public static void InitializeService(IServiceCollection services) =>
+            services.AddRestEaseHttpClient<IAzureAttestationApi>(
+                nameof(IAzureAttestationApi),
+                provider =>
+                {
+                    using var scope = provider.CreateScope();
+
+                    // Can't use HardwareGenerationConfiguration because we are in a singleton lifecycle.

```

Comment:
Amúgy kell, hogy singleton legyen?

Reply:
Úgy emlékszem, hogy igen. Ez már elő jött Finitiveben is. Mármint itt a comment a `services.AddRestEaseHttpClient`-re vonatkozik, nem az `AzureHardwareImplementationComposerBuildProvider`.
Átírtam a kommentet, hogy érthetőbb legyen mi történik.